aten/src/ATen/mps/MPSAllocator.h

```diff
@@ -216,40 +244,59 @@ class MPSHeapAllocatorImpl
   BufferPool m_large_pool_shared, m_large_pool_private;
   // unallocated cached buffers 1 MB or smaller
   BufferPool m_small_pool_shared, m_small_pool_private;
+  // small cached buffers to import scalar values into MPS stream
+  BufferPool m_scalar_pool;
   // total memory allocated by HeapAllocator
   size_t m_total_allocated_memory;
   // max buffer size allowed by Metal
   size_t m_max_buffer_size;
-  // sets a soft upper bound to limit the total allocations
-  bool m_set_fraction;
-  // use "PYTORCH_DEBUG_MPS_ALLOCATOR" env-var to enable debug info
-  bool m_enable_debug_info;
-
+  // maximum total size allowed to be allocated
+  size_t m_max_total_allowed_size;
```

Comment:
Is this total including all the pools ?

Reply:
Yes, all the pools. Currently, this code is not in effect since the high watermark ratio is disabled until we find a proper way to enable it.